src/api/feed-discovery/src/lib.js

```diff
@@ -0,0 +1,85 @@
+const got = require('got');
+const cheerio = require('cheerio');
+const validUrl = require('valid-url');
+const createError = require('http-errors');
+
+// Helper function to return the feed url of a given blog url
+async function getFeedUrls(blogUrl) {
+  const res = await got.get(blogUrl);
+  // Load the html document
+  const $ = cheerio.load(res.body);
+  // Extract the Cheerio object of rss link elements
+  const rss = $('link[type="application/rss+xml"]').clone();
+  // Extract the Cheerio object of atom link elements
+  const atom =
+    $('link[type="application/atom+xml"]').clone() ||
+    $('link[type="application/x.atom+xml"]').clone() ||
+    $('link[type="application/x-atom+xml"]').clone();
+
+  /* NOTE: The return of the clone() function would be a Cheerio object with the
+  keys similar to [ '0', '1', 'options', '_root', 'length', 'prevObject' ]. In this
+  case the length will be 2 since there are two elements with a feed url. */
+
+  const feedUrls = [];
+
+  // If there are rss feeds
+  if (rss.length > 0) {
+    for (let i = 0; i < rss.length; i += 1) {
+      feedUrls.push(rss[i].attribs.href);
+    }
+  }
+  // If there are atom feeds
+  if (atom.length > 0) {
+    for (let i = 0; i < atom.length; i += 1) {
+      feedUrls.push(atom[i].attribs.href);
+    }
+  }
+
+  return feedUrls;
+}
+
+// A middleware to check if Url provided is valid
+function checkValidUrl(req, res, next) {
+  // If the URL is invalid, return 400 error
+  if (!validUrl.isUri(req.body.blogUrl)) {
+    return next(createError(400, 'Invalid Blog URL'));
+  }
+  // Else, continue
+  return next();
+}
+
+// A middleware to check if the URL is a valid web page, 200 status with text/html type
+async function checkValidBlogPage(req, res, next) {
+  try {
+    const { statusCode, headers } = await got.head(req.body.blogUrl);
```

Comment:
Why `.head`?  Can't we get the full thing and pass it along to the next middleware?

Reply:
I am updating it in next commit. Probably later today